### ISSUE TITLE
RFE: support root on LUKS in gpt-auto-generator

### ISSUE THREAD

**Issue by gdamjan** [systemd/systemd#859]
My root filesystem is LUKS encrypted (no LVM, just LUKS on /dev/sda3).

```
dir=$(mktemp -d)
sudo SYSTEMD_LOG_LEVEL=debug ./systemd-gpt-auto-generator $dir $dir $dir
/dev/dm-0: not a partitioned device, ignoring.
```

afterwards $dir is empty. I'd expect at least boot and swap units.

it seems that inside add_mounts(void), get_block_device returns the /dev/md-0 device (/dev/mapper/root) and then gpt-auto-generator calls enumerate_partitions(devno) on the mapper.

**Comment by poettering** [comment]
Yes, gpt-auto-discovery only works on if the root disk is on a GPT partition tables. Hence its name. What did you expect?

**Comment by gdamjan** [comment]
just to make it clear, my disk is GPT partitioned. boot and swap are not LUKS encrypted of course.

http://www.freedesktop.org/wiki/Specifications/DiscoverablePartitionsSpec/ suggests that it shouldn't be different if the partitions are LUKS encrypted.

Also, with the previous EFI boot automount the same partitioning scheme did create the boot mount. That's why expected it to continue to work.

**Comment by poettering** [comment]
Ah, got it now. Actually this has been on the TODO list for a while, see the TODO file in the git repo, look for "gpt-auto-generator" there...

**Comment by poettering** [comment]
Fix waiting in #4879.
